src/cdomains/affineEquality/arrayImplementation/arrayMatrix.ml

```diff
@@ -544,19 +291,14 @@ module ArrayMatrix: AbstractMatrix =
       )
       with Unsolvable -> None
 
-    let rref_matrix_with m1 m2 = timing_wrap "rref_matrix_with" (rref_matrix_with m1) m2
-
-    let normalize_with m =
-      rref_with m
+    let rref_matrix_with m1 m2 = Timing.wrap "rref_matrix_with" (rref_matrix_with m1) m2
 
-    let normalize_with m = timing_wrap "normalize_with" normalize_with m
-
-    let normalize m =
-      let copy = copy m in
-      if normalize_with copy then
-        Some copy
-      else
-        None
+    let rref_matrix m1 m2 = 
+      let m1' = copy m1 in
+      let m2' = copy m2 in 
+      match rref_matrix_with m1' m2' with
+      | Some m -> Some m
+      | None -> None
```

Comment:
What is this match supposed to achieve? It just returns the argument to match. Also why do both need to be copied? Does `rref_matrix_with` modify both arguments?